subsys/bluetooth/audio/tbs.c

```diff
@@ -329,189 +333,182 @@ static uint8_t next_free_call_index(void)
 	return BT_TBS_FREE_CALL_INDEX;
 }
 
-static void net_buf_put_call_state(const void *inst_p)
+static struct bt_tbs_call *call_alloc(struct tbs_service_inst *inst, uint8_t state, const char *uri,
+				      uint16_t uri_len)
+{
+	struct bt_tbs_call *free_call = NULL;
+
+	for (size_t i = 0; i < ARRAY_SIZE(inst->calls); i++) {
+		if (inst->calls[i].index == BT_TBS_FREE_CALL_INDEX) {
+			free_call = &inst->calls[i];
+			break;
+		}
+	}
+
+	if (free_call == NULL) {
+		return NULL;
+	}
+
+	memset(free_call, 0, sizeof(*free_call));
+
+	/* Get the next free call_index */
+	free_call->index = next_free_call_index();
+	if (free_call->index == BT_TBS_FREE_CALL_INDEX) {
+		/* XXX: Consider assert instead */
+		return NULL;
+	}
+
+	free_call->state = state;
+	(void)memcpy(free_call->remote_uri, uri, uri_len);
+	free_call->remote_uri[uri_len] = '\0';
```

Comment:
Should we have an assert or similar to verify that `uri_len < sizeof(remote_uri)`?

Reply:
I'll assert, since the uri_len is already checked by `bt_tbs_valid_uri` function called in `originate_call` and `bt_tbs_remote_incoming`